src/SourceBuild/tarball/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/DotNetHelper.cs

```diff
@@ -41,4 +41,20 @@ public void ExecuteDotNetCmd(string args, ITestOutputHelper outputHelper)
 
         Assert.Equal(0, executeResult.Process.ExitCode);
     }
+
+    /// <Summary>
+    /// Create a new project of type projectType with the specified langauge in the directory outputPath.
+    /// If the directory at outputPath already exists, clear its contents first.
+    /// </Summary>
+    public void NewProject(string projectType, string language, string outputPath, ITestOutputHelper outputHelper)
```

Comment:
I see value in placing all test projects inside a timestamped folder.  This would be similar to what the smoke-test script does today.  This is what I have in another PR I have been working on.

```
   private static readonly string s_timestamp = DateTime.Now.ToString("yyyyMMddHHmmssffff");

    public void ExecuteNew(string projectType, string language, string name, ITestOutputHelper outputHelper)
    {
        string outputDirectory = Path.Combine(Directory.GetCurrentDirectory(), $"projects-{s_timestamp}", name);

        ExecuteCmd($"new {projectType} --language \"{language}\" --name {name} --output {outputDirectory}", outputHelper);
    }
```